Compliance is blocked

### Summary 
Status and result points each other. Message is not clear

### Steps to reproduce the behavior
On an existing eks cluster, with tiller preinstalled 

### Jx version
jx                 1.3.118
kubernetes cluster v1.10.3
kubectl            v1.11.1
helm client        v2.9.1+g20adb27
helm server        v2.9.1+g20adb27
git                git version 2.18.0

The output of `jx version` is:

```
~ ❯❯❯ jx compliance status
Compliance tests have failed. You can check what happened with `jx compliance results`.%                                                                                                                                                      

~ ❯❯❯ jx compliance results
Compliance results not ready. Run `jx compliance status` for status.%                                                                                                                                                                         

~ ❯❯❯ jx compliance logs
namespace="heptio-sonobuoy" pod="sonobuoy" container="kube-sonobuoy"
time="2018-07-27T04:52:48Z" level=info msg="Scanning plugins in ./plugins.d (pwd: /)"
time="2018-07-27T04:52:48Z" level=info msg="Scanning plugins in /etc/sonobuoy/plugins.d (pwd: /)"
time="2018-07-27T04:52:48Z" level=info msg="Directory (/etc/sonobuoy/plugins.d) does not exist"
time="2018-07-27T04:52:48Z" level=info msg="Scanning plugins in ~/sonobuoy/plugins.d (pwd: /)"
time="2018-07-27T04:52:48Z" level=info msg="Directory (~/sonobuoy/plugins.d) does not exist"
time="2018-07-27T04:52:48Z" level=info msg="Filtering namespaces based on the following regex:.*|heptio-sonobuoy"
time="2018-07-27T04:52:48Z" level=info msg="Namespace default Matched=true"
time="2018-07-27T04:52:48Z" level=info msg="Namespace dev Matched=true"
time="2018-07-27T04:52:48Z" level=info msg="Namespace heptio-sonobuoy Matched=true"
time="2018-07-27T04:52:48Z" level=info msg="Namespace kube-public Matched=true"
time="2018-07-27T04:52:48Z" level=info msg="Namespace kube-system Matched=true"
time="2018-07-27T04:52:48Z" level=info msg="Namespace monitoring Matched=true"
time="2018-07-27T04:52:49Z" level=info msg="Starting server Expected Results: [{ e2e} {ip-10-0-35-87.us-west-2.compute.internal systemd_logs} {ip-10-0-39-164.us-west-2.compute.internal systemd_logs} {ip-10-0-50-110.us-west-2.compute.internal systemd_logs}]"
time="2018-07-27T04:52:49Z" level=info msg="Running plugin" plugin=e2e
time="2018-07-27T04:52:49Z" level=info msg="starting aggregation server" address=0.0.0.0 port=8080
time="2018-07-27T04:52:49Z" level=info msg="Running plugin" plugin=systemd-logs
time="2018-07-27T04:52:53Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-39-164.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T04:52:53Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-50-110.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T05:52:55Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-50-110.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T05:52:55Z" level=warning msg="Got a duplicate result systemd_logs/ip-10-0-50-110.us-west-2.compute.internal"
time="2018-07-27T05:52:56Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-39-164.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T05:52:56Z" level=warning msg="Got a duplicate result systemd_logs/ip-10-0-39-164.us-west-2.compute.internal"
time="2018-07-27T06:52:59Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-39-164.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T06:52:59Z" level=warning msg="Got a duplicate result systemd_logs/ip-10-0-39-164.us-west-2.compute.internal"
time="2018-07-27T06:52:59Z" level=info msg="received aggregator request" client_cert=systemd-logs node=ip-10-0-50-110.us-west-2.compute.internal plugin_name=systemd_logs
time="2018-07-27T06:52:59Z" level=warning msg="Got a duplicate result systemd_logs/ip-10-0-50-110.us-west-2.compute.internal"
time="2018-07-27T07:36:47Z" level=info msg="received aggregator request" client_cert=e2e plugin_name=e2e
time="2018-07-27T07:36:47Z" level=info msg="Running non-ns query"
time="2018-07-27T07:36:47Z" level=error msg="error querying CustomResourceDefinitions: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=info msg="Collecting Node Configuration and Health..."
time="2018-07-27T07:36:47Z" level=info msg="Creating host results for ip-10-0-35-87.us-west-2.compute.internal under /tmp/sonobuoy/549619f5-704f-45e0-95c2-fce233d35711/hosts/ip-10-0-35-87.us-west-2.compute.internal\n"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get configz endpoint for node ip-10-0-35-87.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get healthz endpoint for node ip-10-0-35-87.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=info msg="Creating host results for ip-10-0-39-164.us-west-2.compute.internal under /tmp/sonobuoy/549619f5-704f-45e0-95c2-fce233d35711/hosts/ip-10-0-39-164.us-west-2.compute.internal\n"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get configz endpoint for node ip-10-0-39-164.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get healthz endpoint for node ip-10-0-39-164.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=info msg="Creating host results for ip-10-0-50-110.us-west-2.compute.internal under /tmp/sonobuoy/549619f5-704f-45e0-95c2-fce233d35711/hosts/ip-10-0-50-110.us-west-2.compute.internal\n"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get configz endpoint for node ip-10-0-50-110.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=warning msg="Could not get healthz endpoint for node ip-10-0-50-110.us-west-2.compute.internal: the server could not find the requested resource"
time="2018-07-27T07:36:47Z" level=error msg="error querying ThirdPartyResources: don't know how to handle non-namespaced resource ThirdPartyResources"
time="2018-07-27T07:36:47Z" level=info msg="Running ns query (default)"
time="2018-07-27T07:36:49Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:36:50Z" level=info msg="Running ns query (dev)"
time="2018-07-27T07:36:52Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:36:56Z" level=info msg="Running ns query (heptio-sonobuoy)"
time="2018-07-27T07:36:57Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:36:59Z" level=error msg="error querying PodPresets: the server could not find the requested resource (get podpresets.settings.k8s.io)"
time="2018-07-27T07:37:00Z" level=info msg="Running ns query (kube-public)"
time="2018-07-27T07:37:02Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:37:03Z" level=info msg="Running ns query (kube-system)"
time="2018-07-27T07:37:04Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:37:48Z" level=info msg="Running ns query (monitoring)"
time="2018-07-27T07:37:48Z" level=info msg="Collecting Pod Logs..."
time="2018-07-27T07:38:44Z" level=info msg="Results available at /tmp/sonobuoy/201807270452_sonobuoy_549619f5-704f-45e0-95c2-fce233d35711.tar.gz"
time="2018-07-27T07:38:44Z" level=info msg="no-exit was specified, sonobuoy is now blocking"```

### Kubernetes cluster

What kind of Kubernetes cluster are you using & how did you create it?
eks cluster created using https://github.com/terraform-aws-modules/terraform-aws-eks

### Operating system / Environment
mac os 10.13.6

### Expected behavior
should get compliance failed reason

### Actual behavior
Messages circular dependency

I was able to retrieve the results of the Sonobuoy tests by decompressing and unzipping the file mentioned at the end of `jx compress logs`.

You can copy the file to your local machine's root directory with:

```
kubectl cp heptio-sonobuoy/sonobuoy:tmp/sonobuoy/201808301915_sonobuoy_9d655f9c-0a80-4b1f-a635-370a26d5e519.tar.gz /201808301915_sonobuoy_9d655f9c-0a80-4b1f-a635-370a26d5e519.tar.gz
```

To decompress and extract it:

```
gzip -d /201808301915_sonobuoy_9d655f9c-0a80-4b1f-a635-370a26d5e519.tar.gz && tar -xf /201808301915_sonobuoy_9d655f9c-0a80-4b1f-a635-370a26d5e519.tar
```

Now open `plugins/e2e/results/e2e.log` and scroll to the bottom to see the test results. For me:

```
Summarizing 2 Failures:

[Fail] [sig-scheduling] SchedulerPredicates [Serial] [It] validates that NodeSelector is respected if not matching  [Conformance] 
/workspace/anago-v1.11.2-beta.0.50+bb9ffb1654d4a7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:4790

[Fail] [sig-scheduling] SchedulerPredicates [Serial] [It] validates resource limits of pods that are allowed to run  [Conformance] 
/workspace/anago-v1.11.2-beta.0.50+bb9ffb1654d4a7/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:4790

Ran 143 of 996 Specs in 5734.697 seconds
FAIL! -- 141 Passed | 2 Failed | 0 Pending | 853 Skipped --- FAIL: TestE2E (5735.07s)
FAIL

Ginkgo ran 1 suite in 1h35m36.419923376s
Test Suite Failed
```

☹️

--- a/Gopkg.lock
+++ b/Gopkg.lock
@@ -616,7 +616,7 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:5f61e597263e107de9d237f94ed013455acd48d99d662141329f562a5c3a93f5"
+  digest = "1:edca24b46dde248d71ba416f5ff61e878c1c827bbb0ce9e6eced84aac7201c6c"
   name = "github.com/heptio/sonobuoy"
   packages = [
     "pkg/backplane/ca",
@@ -640,7 +640,7 @@
     "pkg/templates",
   ]
   pruneopts = "NUT"
-  revision = "093f4a7fc64edd1c3d1d0b56ba304bf3a050910d"
+  revision = "1063612447506c1f190a11e9c25601cc7243de6c"
 
 [[projects]]
   digest = "1:65300ccc4bcb38b107b868155c303312978981e56bca707c81efec57575b5e06"

--- a/pkg/jx/cmd/compliance_results.go
+++ b/pkg/jx/cmd/compliance_results.go
@@ -76,7 +76,7 @@ func (o *ComplianceResultsOptions) Run() error {
 	}
 
 	if status.Status != aggregation.CompleteStatus {
-		log.Info("Compliance results not ready. Run `jx compliance status` for status.")
+		log.Infoln("Compliance results not ready. Run `jx compliance status` for status.")
 		return nil
 	}
 

--- a/pkg/jx/cmd/compliance_status.go
+++ b/pkg/jx/cmd/compliance_status.go
@@ -64,7 +64,7 @@ func (o *ComplianceStatusOptions) Run() error {
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve the status")
 	}
-	log.Info(hummanReadableStatus(status.Status))
+	log.Infoln(hummanReadableStatus(status.Status))
 	return nil
 }
 

